spring-xd-runner/src/main/java/org/springframework/bus/xd/bootstrap/ModuleOptionsPropertySourceInitializer.java

```diff
@@ -78,9 +80,18 @@ public void initialize(ConfigurableApplicationContext applicationContext) {
 		insert(environment, new MapPropertySource("moduleDefaults", map));
 	}
 
-	private ModuleDefinition getModuleDefinition() {
+	private ModuleDefinition getModuleDefinition(ConfigurableApplicationContext applicationContext) {
+		String location = "file:.";
+		ClassLoader classLoader = applicationContext.getClassLoader();
+		if (classLoader instanceof URLClassLoader) {
+			URL[] urls = ((URLClassLoader) classLoader).getURLs();
+			location = urls[0].toString();
```

Comment:
I think we need to avoid assigning to location here since we don't want to set this unless the next if-condition is true.